### PR TITLE
Typography: new Typography guidelines in the docs

### PR DESCRIPTION
**Pull request by ponori**
Adding type guidelines to Gestalt docs

### CODE REVIEW

**Comment by ponori** [comment]
@ayeshakmaz I'm still working through some errors and adding content, but I thought I'd try to get this party started today with the slim hope that we could get this thing out.

File: docs/pages/typography.js [hunk 1]
```diff
@@ -0,0 +1,1008 @@
...
+        <MainSection.Subsection
+          title="Line height"
+          description="Proper line spacing line height (or leading) is important for readability. If lines of text get too close together, it can be hard to read them; if they are too far apart it can make it difficult to sense a clear hierarchy and group related text blocks. Given that we use system fonts, we rely on percentages on web and system defaults on mobile to ensure blocks are text are still readable for internationalization and dynamic sizing."
+        />
+        <MainSection.Subsection
+          title="Web"
+          description="We currently use browser defaults on web UIs so that lines of text are readable in all
+            languages and scripts. For example, scripts like Hindi have larger ascenders and
+            descenders than Latin scripts. Setting a fixed line height can make scripts like Hindi
+            hard to read."
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:284]: I think you can delete all this (duplicated below)

File: docs/pages/typography.js [hunk 2]
```diff
@@ -0,0 +1,1008 @@
...
+                  <Text size="200">Bold</Text>
+                </Table.Cell>
+                <Table.Cell>
+                  <Text size="200">Semibold</Text>
+                </Table.Cell>
+                <Table.Cell>
+                  <Text size="200">Medium</Text>
+                </Table.Cell>
+                <Table.Cell>
+                  <Heading>Page title</Heading>
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:594]: I'd use Text here with the biggest font size, to pass the a11y test

File: docs/pages/typography.js [hunk 3]
```diff
@@ -0,0 +1,1008 @@
...
+            defaultCode={`
+<Box rounding={2} padding={2} color="infoWeak"><Text>This has enough contrast</Text></Box>
+`}
+          />
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use text that doesn’t pass 4.5:1 contrast ratio when testing with Figma accessibility plugins."
+            defaultCode={`
+<Box rounding={2} padding={2} color="infoWeak"><Text color="inverse">This does not have enough contrast</Text></Box>
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:829]: May need to use an image here due to the a11y tests

File: docs/pages/typography.js [hunk 4]
```diff
@@ -0,0 +1,1008 @@
...
+            cardSize="md"
+            type="do"
+            description="For SEO and users with screen readers, follow a logical hierarchy by using headings based on a numerical order, not on font size."
+          >
+            <HierarchyDo />
+          </MainSection.Card>
+          <MainSection.Card
+            cardSize="md"
+            type="don't"
+            description="Use font sizes below 14px for body copy and UI controls, unless the text is very brief and secondary."
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:810]: This is copy-pasta from above I think, needs to be updated

File: docs/pages/typography.js [hunk 5]
```diff
@@ -0,0 +1,1008 @@
...
+            <Heading size="300" accessibilityLevel={4}>
+              Web
+            </Heading>
+            <Markdown
+              text={`
+We currently use browser defaults on web UIs so that lines of text are readable in all languages and scripts. For example, scripts like Hindi have larger ascenders and descenders than Latin scripts. Setting a fixed line height can make scripts like Hindi hard to read.
+            `}
+            />
+          </Box>
+          <Flex gap={4} width={504}>
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:296]: need to add "wrap" to this flex

File: docs/pages/typography.js [hunk 6]
```diff
@@ -0,0 +1,1008 @@
...
+          </Box>
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Character spacing"
+          description="Character spacing is set to system defaults across all devices and platforms"
+        />
+      </MainSection>
+      <MainSection name="Alignment">
+        <MainSection.Subsection title="Default">
+          <Flex gap={4}>
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:411]: Should add "wrap" here too

File: docs/pages/typography.js [hunk 7]
```diff
@@ -0,0 +1,1008 @@
...
+                    read; this can be especially difficult for people with dyslexia
+                    (https://www.bdadyslexia.org.uk/advice/employers/creating-a-dyslexia-friendly-workplace/dyslexia-friendly-style-guide).
+                  </li>
+                </ul>
+              </Text>
+            </Box>
+          </Flex>
+        </MainSection.Subsection>
+        <MainSection.Subsection title="Exceptions">
+          <Flex gap={4}>
```
Comment (ayeshakmaz) [inline docs/pages/typography.js:442]: i'd add wrap here too

**Comment by AlbertCarreras** [comment]
@ponori The accessibility tests are failing. One has to do with the not-enough contrast example. 


Screen Shot 2022-05-31 at 2 41 40 PM


>> We should disable it or make it an image instead. @ayeshakmaz any thoughts?


Re the second one, there's a H2 missing because there's an incorrect H1.

Screen Shot 2022-05-31 at 2 52 21 PM
Screen Shot 2022-05-31 at 2 51 41 PM

File: docs/pages/typography.js (hunk 2 above)
Comment (AlbertCarreras) [reply docs/pages/typography.js:594]: See comment below

File: docs/pages/typography.js (hunk 3 above)
Comment (AlbertCarreras) [reply docs/pages/typography.js:829]: Yes, see comments below

**Comment by AlbertCarreras** [comment]
You can check the test issues going to Actions https://github.com/pinterest/gestalt/actions
then clicking on your PR title that's failing. https://github.com/pinterest/gestalt/actions/runs/2391875441
Then clicking on [🎭 Playwright Run Summary] You'll see results here
https://github.com/pinterest/gestalt/runs/6612952227?check_suite_focus=true#step:7:89)https://github.com/pinterest/gestalt/runs/6612583336?check_suite_focus=true#step:7:89

File: docs/pages/typography.js (hunk 1 above)
Comment (ponori) [reply docs/pages/typography.js:284]: Thanks! Done!

File: docs/pages/typography.js (hunk 5 above)
Comment (ponori) [reply docs/pages/typography.js:296]: Done!

File: docs/pages/typography.js (hunk 6 above)
Comment (ponori) [reply docs/pages/typography.js:411]: Done!

File: docs/pages/typography.js (hunk 7 above)
Comment (ponori) [reply docs/pages/typography.js:442]: Done!

File: docs/pages/typography.js (hunk 2 above)
Comment (ponori) [reply docs/pages/typography.js]: Done!

File: docs/pages/typography.js (hunk 3 above)
Comment (ponori) [reply docs/pages/typography.js]: Done!

**Review by ayeshakmaz — APPROVED** [review]

File: docs/pages/typography.js [hunk 8]
```diff
@@ -0,0 +1,995 @@
...
+              with users absorbing all the information they need for a good experience
+              - Since the starting edge of centered text moves around, it’s harder for users to
+              read; this can be especially difficult for [people with dyslexia](https://www.bdadyslexia.org.uk/advice/employers/creating-a-dyslexia-friendly-workplace/dyslexia-friendly-style-guide)
+              `}
+              />
+            </Box>
+          </Flex>
+        </MainSection.Subsection>
+        <MainSection.Subsection title="Exceptions">
+          <Flex gap={4} wrap>
```
[the remark this answers is not in the document]
Comment (ponori) [reply docs/pages/typography.js:431]: OK, I made a few more changes to this issue and the other to help address your points.

**Pull request merged**